src/Turtle/Shell.hs

```diff
@@ -78,18 +79,22 @@ import Data.Monoid (Monoid(..), (<>))
 import Data.String (IsString(..))
 
 -- | A @(Shell a)@ is a protected stream of @a@'s with side effects
-newtype Shell a = Shell { foldIO :: forall r . FoldM IO a r -> IO r }
+newtype Shell a = Shell { _foldIO :: forall r . FoldM IO a r -> IO r }
+
+-- | Use a `FoldM IO` to reduce the stream of @a@'s produced by a `Shell`
```

Comment:
Actually, I take that back.  It would be inconsistent with the rest of the progression (each new function is implemented in terms of the previous one).